### PR TITLE
Add Linting

### PR DESCRIPTION
**Pull request by mattruggio**
This PR is in the same vein as the logicality PR. It:

adds linting
addresses all linting warnings and errors
does a little configuration cleanup.
Note this also contains the same breaking change where the removal of un-necessary indirection (namespace) caused a break in API.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,14 @@
...
-  - yarn run build
+- yarn run build
+script:
+- yarn run lint
+- yarn run test
+cache: yarn
+notifications:
+  hipchat:
+    rooms:
+      secure: Q8mJzTZuQGPQJc4xFzdRmqUKVJDLfIxmupLigD3TadNU+j6LTKqyHrYN88bAhw01xY7KxFLVjlvHI0l9/n1GgYVKx1b5CrVUT4khiPygNJKqWAw6nfi/P4LgDlK04fKo0EMgw67lXyYrtmaWtY9MJTYJAv2VyUEWfHHu5tu3dPyJndPAzFHtBCvslO7/+Yf3kidajNCw3VkciArU1786OqapzVN5u1acpMv7h4ls5nJKP3VHZYEqoCAwebccNWc6AS1xjEp4RJYH9Sg29vVmn3EKaFTRpfKewQfieJVgOlEC0+c0QxQFhtGyb73nyBSZQR1Si5hhZnSl3g8ncKUHiUsBBXouSo5oDsODHgySq3b1pLcuHtcT5xmXl/r0eBjf/EASU2rMJ/7m7csXKyEYlB5uZ/IHXgXInnf7gKJ6GD1qrZaFXSMYPGKBQT0F9RkygGzo+13YStv8YrVmLZOmOU33JdfB2S7n/hj4ehXtiA/nUE05bfXLpXjs9OMoMEJrQBeUVVNt0BaIhOU8nB4B2Mxuddjx6ntC1aNfBisa2mQ6G5vkTuaW5qCeahzc4g20lmYEKFsYFXxTNbEkKD+zdxYyF4YPC6kWC6BZZixSUxb2/O/UmNLkd8qDThg3KFFxxQYB79gtik9jgKh0fJSPTvV0Vthj8C42hmfLXgaWE8c=
```
Comment (gerryster) [inline .travis.yml]: Is there another way to configure Travis so this does not have to be exposed to the world?

Comment (mattruggio) [reply .travis.yml]: This is the prescribed way on how to [configure TravisCI for HipChat notifications](https://docs.travis-ci.com/user/notifications/#configuring-hipchat-notifications).

File: README.md [hunk 2]
```diff
@@ -212,20 +216,41 @@ Basic steps to take to get this repository compiling:
...
 ````
-npm run test
+yarn run lint
 ````
 
+### Publishing
+
+Note: ensure you have proper authorization before trying to publish new versions.
+
+After code changes have successfully went through the Pull Request review process then the following steps should be followed for publishing new versions:
```
Comment (gerryster) [inline README.md]: Grammatically, I would use gone instead of went: "After code changes have successfully _gone_ through..."

Comment (mattruggio) [reply README.md]: I updated this.

**Pull request merged**